src/views/tutorial-view/server.ts

```diff
@@ -0,0 +1,72 @@
+import { Product as ProductContext } from 'types/products'
+import { getAllCollections } from 'lib/learn-client/api/collection'
+import { getTutorial } from 'lib/learn-client/api/tutorial'
+import {
+  ProductOption,
+  TutorialFullCollectionCtx as ClientTutorial,
+} from 'lib/learn-client/types'
+import { stripUndefinedProperties } from 'lib/strip-undefined-props'
+import { splitProductFromFilename } from './helpers'
+
+// @TODO just a stub - adjust page props interface
+export interface TutorialPageProps {
+  tutorial: ClientTutorial
+  product: Pick<ProductContext, 'slug'>
```

Comment:
Relevant for when you decide to get the `ProductSwitcher` working for these pages: this will need to be a `Product` object with more than a `slug` defined. `ProductSwitcher` also needs this object to have a `name`, but we've passed whole product objects (defined in `src/data/<product-slug>.json`) for the `product` prop passed to page components because it's how the `CurrentProductContext` is initialized in `src/pages/_app.tsx`.